xpath.d.ts

```diff
@@ -38,14 +38,14 @@ export function selectWithResolver(expression: string, node: Node, resolver: XPa
 export function useNamespaces(namespaceMap: Record<string, string>): XPathSelect;
 
 // Type guards to narrow down the type of the selected type of a returned Node object
-export function isNodeLike(value: SelectedValue): value is Node;
-export function isArrayOfNodes(value: SelectedValue): value is Node[];
-export function isElement(value: SelectedValue): value is Element;
-export function isAttribute(value: SelectedValue): value is Attr;
-export function isTextNode(value: SelectedValue): value is Text;
-export function isCDATASection(value: SelectedValue): value is CDATASection;
-export function isProcessingInstruction(value: SelectedValue): value is ProcessingInstruction;
-export function isComment(value: SelectedValue): value is Comment;
-export function isDocumentNode(value: SelectedValue): value is Document;
-export function isDocumentTypeNode(value: SelectedValue): value is DocumentType;
-export function isDocumentFragment(value: SelectedValue): value is DocumentFragment;
+export function isNodeLike(value: SelectSingleReturnType): value is Node;
+export function isArrayOfNodes(value: SelectReturnType): value is Node[];
+export function isElement(value: SelectSingleReturnType): value is Element;
+export function isAttribute(value: SelectSingleReturnType): value is Attr;
+export function isTextNode(value: SelectSingleReturnType): value is Text;
+export function isCDATASection(value: SelectSingleReturnType): value is CDATASection;
+export function isProcessingInstruction(value: SelectSingleReturnType): value is ProcessingInstruction;
+export function isComment(value: SelectSingleReturnType): value is Comment;
+export function isDocumentNode(value: SelectSingleReturnType): value is Document;
+export function isDocumentTypeNode(value: SelectSingleReturnType): value is DocumentType;
+export function isDocumentFragment(value: SelectSingleReturnType): value is DocumentFragment;
```

Comment:
In testing this in `xml-crypto` it seems that it would be helpful to take in `null` and `undefined` as arguments, this way I don't have to have multiple checks.

```suggestion
export function isNodeLike(value?: SelectSingleReturnType | null): value is Node;
export function isArrayOfNodes(value?: SelectReturnType | null): value is Node[];
export function isElement(value?: SelectSingleReturnType | null): value is Element;
export function isAttribute(value?: SelectSingleReturnType | null): value is Attr;
export function isTextNode(value?: SelectSingleReturnType | null): value is Text;
export function isCDATASection(value?: SelectSingleReturnType | null): value is CDATASection;
export function isProcessingInstruction(value?: SelectSingleReturnType | null): value is ProcessingInstruction;
export function isComment(value?: SelectSingleReturnType | null): value is Comment;
export function isDocumentNode(value?: SelectSingleReturnType | null): value is Document;
export function isDocumentTypeNode(value?: SelectSingleReturnType | null): value is DocumentType;
export function isDocumentFragment(value?: SelectSingleReturnType | null): value is DocumentFragment;
```

Reply:
Would you prefer a new type of something like `type Nullable<T> = T | null` with `Nullable<SelectReturnType>`?

Reply:
Actually, it occurred to me that we can probably just take `any` and narrow from there.